hotsos/core/plugins/kernel/kernlog/calltrace.py

```diff
@@ -64,15 +61,15 @@ def __repr__(self):
         s = []
         for k, v in self._info.items():
             if k == 'nodes':
-                s.append("{}{}:".format(prefix, k))
+                s.append(f"{prefix}{k}:")
                 for node, zones in v.items():
-                    s.append(">> node{}".format(node))
+                    s.append(f">> node{node}")
                     for zone, fields in zones.items():
-                        s.append(" {}: {}".format(zone, repr(fields)))
+                        s.append(f" {zone}: {repr(fields)}")
             else:
-                s.append("{}{}: {}".format(prefix, k, v))
+                s.append(f"{prefix}{k}: {v}")
 
-        return "\n{}\n".format('\n'.join(s))
+        return "\n{}\n".format('\n'.join(s))  # noqa, pylint: disable=consider-using-f-string
```

Comment:
Why we're suppressing the warning here?

```
f"\n{'\n'.join(s)}\n"
```
should behave the same:

```python
>>> s=['this', 'is', 'a', 'test']
>>> "\n{}\n".format('\n'.join(s))
'\nthis\nis\na\ntest\n'
>>> f"\n{'\n'.join(s)}\n"
'\nthis\nis\na\ntest\n'

```

Reply:
it failed when I tried it but perhaps I did it wrong, ill try again.

Reply:
nevermind, it's an error in py3.12 but supported in >=3.12, hence the confusion.

Reply:
so yeah, not supported < py3.12 https://peps.python.org/pep-0701/

Reply:
guess we'll have to stick with disables until we grow up to be >= py312 ;)

Reply:
there's a workaround though:

```python
 s_w_newlines = '\n'.join(s)
 return "\n{s_w_newlines}\n"
```